File: Task3/authentication.c
#include <stdio.h>
#include <string.h>

int main()
{
    char username[30];
    char password[30];

    char correctUsername[] = "admin";
    char correctPassword[] = "password123";

    printf("===== User Authentication =====\n\n");

    printf("Enter Username: ");
    scanf("%s", username);

    printf("Enter Password: ");
    scanf("%s", password);

    if(strcmp(username, correctUsername) == 0 &&
       strcmp(password, correctPassword) == 0)
    {
        printf("\nLogin Successful!\n");
        printf("Welcome %s\n", username);
    }
    else
    {
        printf("\nLogin Failed!\n");
        printf("Invalid username or password.\n");
    }

    return 0;
}

File: Task3/authentication.py
"""
ST5004CEM Operating Systems and Security
Task 3 - User Authentication

Author: Trisha Rajbanshi
"""

users = {
    "admin": "admin123",
    "student": "student123"
}


def login():
    print("=" * 50)
    print("USER AUTHENTICATION SYSTEM")
    print("=" * 50)

    username = input("Enter Username: ")
    password = input("Enter Password: ")

    if username in users:
        if users[username] == password:
            print("\nLogin Successful!")
            print(f"Welcome, {username}")
        else:
            print("\nIncorrect Password.")
    else:
        print("\nUser does not exist.")


if __name__ == "__main__":
    login()
